Checker Framework crashes on a reflective call

**Summary**

The Checker Framework has crashed on an Maven compilation on [this project and branch](https://github.com/jyoo980/Chronicle-Core/tree/yoo/checkerframework-crash).

I am able to reproduce this behaviour for the Nullness Checker and the Optional Checker, and this occurs on the latest version (3.40.0) of the Checker Framework.

**Steps to Reproduce**

Please run the following sequence of commands on your shell.
```sh
# Clone the repository and cd into it
git clone git@github.com:jyoo980/Chronicle-Core.git
cd Chronicle-Core

# Check out the branch with the pom.xml changes required to execute the Checker Framework
git checkout yoo/checkerframework-crash
mvn -X compile
```

**Output**

The entire log is much too large to view inline (but is attached), but the relevant stack trace is:

```
[ERROR]   ; The Checker Framework crashed.  Please report the crash.  Version: Checker Framework 3.40.0.
[ERROR]   Compilation unit: /Users/yoo/workspace/Chronicle-Core/src/main/java/net/openhft/chronicle/core/internal/ClassUtil.java
[ERROR]   Last visited tree at line 71 column 97:
[ERROR]                   boolean newFlag = (boolean) setAccessible0_Method.invokeExact(accessibleObject, true);
[ERROR]   Exception: java.lang.Throwable; java.lang.Throwable
[ERROR]   	at org.checkerframework.javacutil.BugInCF.<init>(BugInCF.java:22)
[ERROR]   	at org.checkerframework.framework.type.visitor.AtmComboVisitor.defaultAction(AtmComboVisitor.java:58)
[ERROR]   	at org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visitPrimitive_Array(AbstractAtmComboVisitor.java:387)
[ERROR]   	at org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:504)
[ERROR]   	at org.checkerframework.framework.type.DefaultTypeHierarchy.isSubtype(DefaultTypeHierarchy.java:258)
[ERROR]   	at org.checkerframework.framework.type.DefaultTypeHierarchy.isSubtype(DefaultTypeHierarchy.java:136)
[ERROR]   	at org.checkerframework.common.basetype.Ba
seTypeVisitor.commonAssignmentCheck(BaseTypeVisitor.java:3055)
[ERROR]   	at org.checkerframework.checker.nullness.NullnessVisitor.commonAssignmentCheck(NullnessVisitor.java:250)
[ERROR]   	at org.checkerframework.common.basetype.BaseTypeVisitor.checkVarargs(BaseTypeVisitor.java:1938)
[ERROR]   	at org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:1805)
[ERROR]   	at org.checkerframework.checker.nullness.NullnessVisitor.visitMethodInvocation(NullnessVisitor.java:554)
[ERROR]   	at org.checkerframework.checker.nullness.NullnessVisitor.visitMethodInvocation(NullnessVisitor.java:70)
...
```

Error log: [error.log](https://github.com/typetools/checker-framework/files/13246291/error.log)


**Expectation**

I would not expect the Checker Framework to crash.

--- a/framework/tests/all-systems/Issue6282.java
+++ b/framework/tests/all-systems/Issue6282.java
@@ -1,0 +1,20 @@
+import java.lang.invoke.MethodHandle;
+import java.lang.reflect.AccessibleObject;
+
+public class Issue6282 {
+  public static final MethodHandle setAccessible0_Method = setAccessible0_Method();
+
+  public static final MethodHandle setAccessible0_Method() {
+    throw new RuntimeException();
+  }
+
+  public static void setAccessible(final AccessibleObject accessibleObject) {
+
+    try {
+      boolean newFlag = (boolean) setAccessible0_Method.invokeExact(accessibleObject, true);
+      assert newFlag;
+    } catch (Throwable throwable) {
+      throw new AssertionError(throwable);
+    }
+  }
+}

--- a/framework/tests/all-systems/Issue6282.java
+++ b/framework/tests/all-systems/Issue6282.java
@@ -1,6 +1,7 @@
 import java.lang.invoke.MethodHandle;
 import java.lang.reflect.AccessibleObject;
 
+@SuppressWarnings("ainfertest") // only check WPI for crashes
 public class Issue6282 {
   public static final MethodHandle setAccessible0_Method = setAccessible0_Method();
 

--- a/javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java
+++ b/javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java
@@ -732,15 +732,15 @@ public final class TreeUtils {
           type);
     }
     ExecutableType executableType = (ExecutableType) type;
-    if (((ExecutableType) type).getParameterTypes().isEmpty() && elementFromUse(tree).isVarArgs()) {
+    ExecutableElement element = elementFromUse(tree);
+    if (((ExecutableType) type).getParameterTypes().size() != element.getParameters().size()) {
       // Sometimes when the method type is viewpoint-adapted, the vararg parameter disappears,
       // just return the declared type.
       // For example,
       // static void call(MethodHandle methodHandle) throws Throwable {
       //   methodHandle.invoke();
       // }
-      ExecutableElement ele = elementFromUse(tree);
-      return (ExecutableType) ele.asType();
+      return (ExecutableType) element.asType();
     }
     return executableType;
   }

